### PR TITLE
Adding provider support for Drift Schedules and Ttl Schedules

### PR DESCRIPTION
**Pull request by IaroslavTitov**
### Summary
- Adding schema for drift and ttl schedules
- Adding pulumi service resources for both
- Extended schedule client to be able to call C and U operations for drift and schedules
- Moved some functionality into shared functions to reuse between raw, drift and ttl schedules

### Testing
- Tested locally using DotNet SDK
- Creating, Updating, Destroying works as expected
- Verified in console schedules are created and showing up
```
    var drift = new DriftSchedule(
        "Drift Schedule 1",
        new DriftScheduleArgs{
            Organization = "IaroslavTitov",
            Project = "PulumiDotnet",
            Stack = "SdkTest3",
            ScheduleCron = "0 8 * * 0",
            AutoRemediate = false
        }
    );

    var ttl = new TtlSchedule(
        "Ttl Schedule 1",
        new TtlScheduleArgs{
            Organization = "IaroslavTitov",
            Project = "PulumiDotnet",
            Stack = "SdkTest3",
            Timestamp = "2026-01-01T00:00:00Z",
            DeleteAfterDestroy = true
        }
    );
    ```

### CODE REVIEW

File: provider/cmd/pulumi-resource-pulumiservice/schema.json [hunk 1]
```diff
@@ -1062,6 +1062,130 @@
...
+          "description": "Schedule ID of the created schedule, assigned by Pulumi Cloud.",
+          "type": "string"
+        }
+      },
+      "required": [
+        "organization",
+        "project",
+        "stack",
+        "timestamp",
+        "scheduleID"
```
[the remark this answers is not in the document]
Comment (IaroslavTitov) [reply provider/cmd/pulumi-resource-pulumiservice/schema.json:1159]: In pulumi-cloud repo, lint complains about camelcase and forces ID (which I don't like actually lol)
So I thought that was the convention

[the remark this answers is not in the document]
Comment (komalali) [reply provider/cmd/pulumi-resource-pulumiservice/schema.json:1159]: Heh, yeah fair confusion. In this schema (and I think pulumi schema in general) the convention is camelcase.

File: provider/pkg/internal/pulumiapi/schedules.go [hunk 2]
```diff
@@ -29,6 +33,16 @@ type CreateDeploymentScheduleRequest struct {
 	Request      CreateDeploymentRequest `json:"request,omitempty"`
 }
 
+type CreateDriftScheduleRequest struct {
+	ScheduleCron  *string `json:"scheduleCron,omitempty"`
+	AutoRemediate bool    `json:"autoRemediate,omitempty"`
+}
+
+type CreateTtlScheduleRequest struct {
+	Timestamp          *time.Time `json:"timestamp,omitempty"`
```
[the remark this answers is not in the document]
Comment (IaroslavTitov) [reply provider/pkg/internal/pulumiapi/schedules.go:42]: I thought there was a reason when I was writing it, but as I'm looking at it now I can't think of it haha
I'll make it not a pointer, same for the ScheduleCron in Drift

File: provider/cmd/pulumi-resource-pulumiservice/schema.json [hunk 3]
```diff
@@ -1062,6 +1062,130 @@
...
+        },
+        "timestamp": {
+          "description": "When to run the destroy command, in ISO format like this 2020-01-01T00:00:00Z.",
+          "type": "string"
+        },
+        "scheduleID": {
+          "description": "Schedule ID of the created schedule, assigned by Pulumi Cloud.",
+          "type": "string"
+        }
+      },
```
[the remark this answers is not in the document]
Comment (IaroslavTitov) [reply provider/cmd/pulumi-resource-pulumiservice/schema.json:1153]: Great catch, adding

File: provider/cmd/pulumi-resource-pulumiservice/schema.json [hunk 4]
```diff
@@ -1062,6 +1062,130 @@
...
+        "stack": {
+          "description": "Stack name.",
+          "type": "string"
+        },
+        "timestamp": {
+          "description": "When to run the destroy command, in ISO format like this 2020-01-01T00:00:00Z.",
+          "type": "string"
+        },
+        "deleteAfterDestroy": {
+          "description": "Whether the stack should be deleted after it is destroyed.",
```
[the remark this answers is not in the document]
Comment (IaroslavTitov) [reply provider/cmd/pulumi-resource-pulumiservice/schema.json:1179]: I copypasted this from `pulumiservice:index:OperationContextOptions` :D 
But I'll update to make it fit the rest API docs

**Review by komalali — APPROVED** [review]

**Pull request merged**